extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayDataSourceRuntimeConfig.java

```diff
@@ -131,6 +131,12 @@ public static FlywayDataSourceRuntimeConfig defaultConfig() {
     @ConfigItem
     public boolean cleanDisabled;
 
+    /**
+     * true to automatically call clean when a validation error occurs, false otherwise.
+     */
+    @ConfigItem
+    public boolean cleanOnValidationError;
```

Comment:
With the current implementation, this will only apply to validation at start, not to explicit validation when the code retrieves a `Flyway` object and calls `validate` on it.

So... This should probably be under `validate-at-start`?

Here's a suggestion:

```suggestion
    /**
     * true to automatically execute a Flyway clean command when a validation error occurs at start, false otherwise.
     */
    @ConfigItem(name = "validate-at-start.clean-on-validation-error")
    public boolean cleanOnValidationError;
```

Obviously the test will need to change as well.

Reply:
OK but then you have to configure validate-at-start in YAML as follows:

```yaml
quarkus:
  flyway:
    validate-at-start:
      ~: true
      clean-on-validation-error: true
```

Reply:
Yes.

Alternatively, you could call this `clean-on-validate-at-start-error`? IDK.

Reply:
I#ve added `validate-at-start.clean-on-validation-error` as suggested (see ca5069d095464c055245862bbe5c76630bb4d3c2).